diff --git a/node_modules/@imgly/background-removal/dist/index.mjs b/node_modules/@imgly/background-removal/dist/index.mjs
index 8394ba4..cf8d904 100644
--- a/node_modules/@imgly/background-removal/dist/index.mjs
+++ b/node_modules/@imgly/background-removal/dist/index.mjs
@@ -5889,7 +5889,7 @@ var threads = () => (async (e) => {
   ])
 );
 var webgpu = () => navigator.gpu !== void 0;
-var maxNumThreads = () => navigator.hardwareConcurrency ?? 4;
+var maxNumThreads = () => navigator.hardwareConcurrency ?? 1;
 
 // src/onnx.ts
 async function createOnnxSession(model, config) {
@@ -5907,7 +5907,7 @@ async function createOnnxSession(model, config) {
     ort.env.debug = true;
     ort.env.logLevel = "verbose";
   }
-  ort.env.wasm.numThreads = maxNumThreads();
+  ort.env.wasm.numThreads = config.device === "gpu" ? maxNumThreads() : (navigator.hardwareConcurrency >= 4 ? 2 : 1);
   ort.env.wasm.simd = simd();
   ort.env.wasm.proxy = proxyToWorker;
   const wasmPaths = {
